script.js

```diff
@@ -107,174 +23,196 @@ document.addEventListener("DOMContentLoaded", function () {
+
+  const descriptionInput = getElById("actionModalDescription");
+  descriptionInput.addEventListener("input", updateLiveCodePreview);
+
+  // Add a change event listenter to each key value input
+  const propertyInputs = document.querySelectorAll(
+    "#actionModal input.form-control"
+  );
+  propertyInputs.forEach((input) => {
+    input.addEventListener("change", updateLiveCodePreview);
+  });
+
+  //run the code Preview for the first time
+  updateLiveCodePreview();
+}
+
+function generatePropertyRow(property) {
+  const propertyRow = document.createElement("div");
+  propertyRow.classList.add("row", "mb-3", "align-items-center"); // Add align-items-center and adjust margin
+
+  const propertyNameCol = document.createElement("div");
+  propertyNameCol.classList.add("col-md-4", "fw-bold", "text-end"); // Left-justify the label
+
+  const propertyName = document.createElement("span");
+  propertyName.textContent = property.name;
+
+  propertyNameCol.appendChild(propertyName);
+
+  const propertyValueCol = document.createElement("div");
+  propertyValueCol.classList.add("col-md-8"); // Use the rest of the space
+
+  const propertyValueInput = document.createElement("input");
+  propertyValueInput.type = property.type;
+  propertyValueInput.id = property.id;
+  propertyValueInput.value = property.value;
+  propertyValueInput.setAttribute("data-property-name", property.name); // Add data attribute
+  propertyValueInput.classList.add("form-control"); // Apply Bootstrap styling
+
+  propertyValueCol.appendChild(propertyValueInput);
+
+  propertyRow.appendChild(propertyNameCol);
+  propertyRow.appendChild(propertyValueCol);
+
+  return propertyRow;
+}
+
+// Action MODAL Display Functionality
+function generateActionModalContent(action) {
+  const modalTitle = getElById("actionModalLabel");
+  modalTitle.textContent = action.title;
 
-  // ACTION -LEARN MORE- MODAL FUNCTIONALITY
-  const modalTriggerButtons = document.querySelectorAll(".btn-learn-more");
-  const modalTitle = document.getElementById("actionModalLabel");
   const modalImage = document.querySelector(".modal-body .modal-image");
+  modalImage.src = action.image;
+  modalImage.alt = action.title;
+
   const modalDescription = document.querySelector(
     ".modal-body .modal-description"
   );
-  const actionUsModal = new bootstrap.Modal(
-    document.getElementById("actionModal")
-  );
-  const segmentButton = document.getElementById("segmentButton"); // Move it here
+  modalDescription.textContent = action.description;
+
+  const modalForm = document.querySelector("#actionModal .action-content");
+  modalForm.innerHTML = ""; // Clear previous fields
+
+  if (action.properties.length > 0) {
+    const propertiesContainer = document.createElement("div");
+    propertiesContainer.classList.add("mt-4"); // Add margin-top for separation
+
+    action.properties.forEach((property) => {
+      const propertyRow = generatePropertyRow(property);
```

Comment:
i moved the chunk of code that was here to its own function -- simply just to make each overall function more bite-size